General fixes

milljm opened the pull request:
Quality of Life improvements (try_command for multiple attempts before dying)
Fixes to PETSc patches (to make this work with latest LLVM).
Enable JIT in GCC, RTTI in LLVM.
Added feature to allow user to specify the temp directory to build in.

Commit:
GCC/LLVM latest release, fix JIT for Darwin

Move to the latest release for both GCC and LLVM.
Patch GCC to work with darwin when building jit language addition.
Conform to shell standards when using equal signs during comparisons.

Commit:
Patch PETSc config

Patching PETSc config for Darwin targets. There is an issue
when linking C and Fortran using latest LLVM (5.0)

We need to scrutinize this heavily before allowing this to
be merged...

Commit:
General fixes to 'try_command'

Fixed issue where try_command was being overridden by other
functions wanting to exit. With this change, I have moved all
the PETSc building back to parallel builds.

Enabled 'try_command' for just about anything network related.

Fix the patching files created for the PETSc 3.6.4 targets.

Enable RTTI in LLVM

Closes #114 #113

moosebuild commented:
Job [OpenSUSE 42 Package Tester](https://www.moosebuild.org/job/128625/) on c79faf7 : invalidated by @milljm

Test moose test parallel: *** Error in `python2.7&#39;: free(): invalid pointer: 0x00007f3be0000b20 ***, and then the tester assumed `exodiff` failed. But exodiff never ran (according to stdout). interesting.

moosebuild commented:
Job [Fedora 25 Package Tester](https://www.moosebuild.org/job/128622/) on c79faf7 : invalidated by @milljm

I can run this failing test manually without issues. Trying again. heat_conduction: test:restart.restart_3

Commit:
Allow Trilinos to use PETSc's blas/lapack

Trilinos is linking to system fortran, which may be
very old, causing conflicts when using system blas
(which would have been built with these old compilers)

Closes #114 #113

moosebuild commented:
Job [High Sierra Package Builder](https://www.moosebuild.org/job/129809/) on 726fcbc : invalidated by @milljm

Probably wasn&#39;t done downloading files during previous build

moosebuild commented:
Job [CentOS 7 Package Tester](https://www.moosebuild.org/job/129884/) on ff981ca : invalidated by @milljm

vectorpostprocessors/csv_reader.read: Tester failed, reason: CSVDIFF

moosebuild commented:
Job [Fedora 25 Package Tester](https://www.moosebuild.org/job/129885/) on ff981ca : invalidated by @milljm

try again. Weird issues when the other VMs are busy I think...

Commit:
Continue to use GCC 6.2.0

We are not ready to support GCC 7.2.0, so switch back to 6.2.0.
Update Trilinos/DTK as that was getting dated.

Refs #113
Closes #114

moosebuild commented:
Job [Mint 18 Package Tester](https://www.moosebuild.org/job/130478/) on 4f609f7 : invalidated by @milljm

works when I manually run these tests... (Test moose test python segfault *** Error in `python2.7&#39;: free(): invalid pointer: 0x00007f08c0000900 ***) Possible out of memory?

moosebuild commented:
Job [Ubuntu 14 Package Tester](https://www.moosebuild.org/job/130481/) on 4f609f7 : invalidated by @milljm

worked fine when I manually ran: ./run_tests -j 4 --n-threads=2

moosebuild commented:
Job [Fedora 25 Package Tester](https://www.moosebuild.org/job/130476/) on 4f609f7 : invalidated by @milljm

distributed mesh passes when run manually. Trying again. However, we know test:restart.restart_3 test will fail in modules due to some crazy bug that only happens on Fedora.

Commit:
git clone Trilinos and DTK instead of tarball

Allow the system to git clone a release tag, rather than
use a tarball.

Refs #113
Closes #114

moosebuild commented:
Job [CentOS 7 Package Builder](https://www.moosebuild.org/job/130884/) on cd523e7 : invalidated by @milljm

miniconda failed to connect to its repo and install packages...

moosebuild commented:
Job [Ubuntu 16 Package Builder](https://www.moosebuild.org/job/130892/) on cd523e7 : invalidated by @milljm

miniconda failed to connect to its repo and install packages...

moosebuild commented:
Job [Fedora 25 Package Builder](https://www.moosebuild.org/job/130885/) on cd523e7 : invalidated by @milljm

stupid kerberos authentication...

moosebuild commented:
Job [Mint 18 Package Builder](https://www.moosebuild.org/job/130887/) on cd523e7 : invalidated by @milljm

stupid kerberos authentication...

moosebuild commented:
Job [OpenSUSE 42 Package Builder](https://www.moosebuild.org/job/130888/) on cd523e7 : invalidated by @milljm

stupid kerberos authentication...

moosebuild commented:
Job [Ubuntu 14 Package Builder](https://www.moosebuild.org/job/130891/) on cd523e7 : invalidated by @milljm

stupid kerberos authentication...

moosebuild commented:
Job [Mint 18 Package Builder](https://www.moosebuild.org/job/130887/) on cd523e7 : invalidated by @milljm

what??? /home/moosetest/civet/client_root_0/package_builder/build_from_source/packages/cppunit-gcc: line 27: /opt/moose/Modules/3.2.10/init/bash: No such file or directory

moosebuild commented:
Job [Fedora 25 Package Builder](https://www.moosebuild.org/job/130885/) on cd523e7 : invalidated by @milljm

Dirty build...

moosebuild commented:
Job [Fedora 25 Package Builder](https://www.moosebuild.org/job/130885/) on cd523e7 : invalidated by @milljm

try again after client restart

moosebuild commented:
Job [Fedora 25 Package Builder](https://www.moosebuild.org/job/130885/) on cd523e7 : invalidated by @milljm

killing it on purpose, to see if it can recover properly

moosebuild commented:
Job [Ubuntu 14 Package Tester](https://www.moosebuild.org/job/130910/) on cd523e7 : invalidated by @milljm

try again

moosebuild commented:
Job [CentOS 7 Package Tester](https://www.moosebuild.org/job/130904/) on cd523e7 : invalidated by @milljm

try again

moosebuild commented:
Job [CentOS 7 Package Tester](https://www.moosebuild.org/job/130904/) on cd523e7 : invalidated by @milljm

try again

Commit:
More optimizations

Allow miniconda to try several times before failing.

Fix an odd issue that openSUSE is just now starting to develop, where
the lib directory of built dependencies for icecream, or created as
lib64. I have no idea why this is just now starting to happen.

Commit:
More Optimizations

Move back on up to GCC 7.2.0
Build our own BLAS LAPACK shared library for use with
Trilinos.

Use the substitution template feature as much as possible. This
allows for path/naming conventions to be easily maintained.

Add environmental variables to each package so they are self
containing. Which allows:

Add an interactive debug mode feature for individual packages.

Update the way ccache works while building the package.

Prevent package 'valgrind' from using ccache.

moosebuild commented:
Job [CentOS 7 Package Builder](https://www.moosebuild.org/job/132680/) on d263114 : canceled by @milljm

problem with ccache

moosebuild commented:
Job [Mint 18 Package Builder](https://www.moosebuild.org/job/132683/) on d263114 : canceled by @milljm

problem with ccache

Commit:
Syntax and Build fixes

Do not build GCC/Clang with CCACHE enabled. This seems to break
Darwin targets when building anything requires clang to link to
fortran objects.

Fix module load typo in petsc-default-mpich-clang-opt.

Fix icecream_stack; OpenSUSE insists on making lib64 directories.

Add ability to override jobs when running scripts interactively.

Closes #114 #115

Commit:
Disable CCACHE when building the package

We are after stability, not speed, when building the
package.

Closes #134 #115

moosebuild commented:
Job [High Sierra Package Creator](https://www.moosebuild.org/job/133849/) on 8754855 : invalidated by @milljm

clean repo, try again

Commit:
Multiple issue fix...

Fix a rather nasty bug that was causing any failing build, to pass!
Update Valgrind to work in parallel.

Closes #114

Commit:
Support Possible spaces in files/directories

Fix cario not copying its pkg-config file.

Commit:
Fix Valgrind module

Valgrind required a dependency that was not present (mpich-gcc).
Added libperl-dev to our Linux build machines as a requirement.

Closes #114

Commit:
Fix Trilinos blas/lapack library path

Just give Trilinos the directory path and let it
choose the blas/lapack libraries to load.

Closes #114

Commit:
Revert back to Trilinos 12.6.2

Revert to Trilinos 12.6.2 and use PETSc's blas/lapack
libraries. We will get a 'libfblas/lapack not portable
warning', but I can live with that.

Modified how several Trilnos civet modules were created.

Closes #114

Commit:
Trilinos on Darwin to use system blas/lapack

Darwin will continue to use system blas/lapack when
building Trilinos.

Closes #114 #115

milljm commented:
Lets do this. So we can start to support High Sierra.
